cpanspec

```diff
@@ -1905,4 +1896,50 @@ sub dump_statistics {
     verbose($yaml);
 }
 
+sub filter_requires {
+    my ($name) = @_;
+    my $script="$name-filter-requires.sh";
+    verbose "Writing $script...";
+    my $sh;
+    if ($force) {
+        rename($script, "$script~") if (-e $script);
+        $sh=new FileHandle ">$script";
+    } else {
+        $sh=new FileHandle $script, O_WRONLY|O_CREAT|O_EXCL;
+    }
+    die "Failed to create $script: $!\n" if (!$sh);
+
+    print $sh "#!/bin/sh\n\n"
+        . "\@\@PERL_REQ\@\@ \"\$\@\" | sed -e '/^$filter_requires[0]\$/d'";
+    if (@filter_requires > 1) {
+        for my $dep (@filter_requires[1..$#filter_requires]) {
+            print $sh " \\\n    -e '/^$dep\$/d'";
+        }
+    }
+    print $sh "\n";
+}
+
+sub filter_provides {
+    my ($name) = @_;
+    my $script = "$name-filter-provides.sh";
+    verbose "Writing $script...";
+    my $sh;
+    if ($force) {
+        rename($script, "$script~") if (-e $script);
+        $sh = new FileHandle ">$script";
+    }
+    else {
+        $sh = new FileHandle $script, O_WRONLY | O_CREAT | O_EXCL;
+    }
+    die "Failed to create $script: $!\n" if (!$sh);
+
+    print $sh "#!/bin/sh\n\n" . "\@\@PERL_PROV\@\@ \"\$\@\" | sed -e '/^$filter_provides[0]\$/d'";
+    if (@filter_provides > 1) {
+        for my $dep (@filter_provides[1 .. $#filter_provides]) {
```

Comment:
If `@filter_provides` isn't used anymore one you just `shift`-away the first element and use a simple loop.

Reply:
I don't understand that sentence. But I also have no idea what this code and the --filter-provides option is doing at all and I just moved it into a subroutine (so I can forget about it ;-)

Reply:
Considering the lack of tests, I second leaving it. Simplifying can backfire if you don't know the code